### PR TITLE
Fix binaries build

### PR DESCRIPTION
**Pull request by evanxg852000**
Fixes builds:
- x86_64-unknown-linux-gnu
- aarch64-unknown-linux-gnu

Commented both `musl` builds: 
There seem to be conflicting statements about RocksDB compilation with musl (possible/impossible). 
An investigation is being carried out in the [branch](https://github.com/quickwit-oss/quickwit/tree/fix-1219-musl). The build step is ok but the linking step fails. I suspect this can take a bit longer so let's have the other build and our release workflow at least working.

Part of https://github.com/quickwit-oss/quickwit/issues/1219

### CODE REVIEW

File: build/cross-images/x86_64-unknown-linux-musl.dockerfile [hunk 1]
```diff
@@ -11,3 +11,5 @@ RUN echo "Upgrading CMake" && \
 ENV CC=musl-gcc \
     CFLAGS=-I/usr/local/musl/include \
     LIB_LDFLAGS=-L/usr/lib/x86_64-linux-gnu
+
+RUN sudo chown -R rust:rust /opt/rust/
```
Comment (guilload) [inline build/cross-images/x86_64-unknown-linux-musl.dockerfile:15]: Just curious, why is this needed? Also, do you need `sudo`, I believe docker containers are built and run using the user `root`.

File: build/cross-images/aarch64-unknown-linux-gnu.dockerfile [hunk 2]
```diff
@@ -4,7 +4,8 @@ COPY libpq.sh /
 RUN bash /libpq.sh aarch64-linux-gnu
 
 RUN apt-get update -y && \
-    apt-get install -y libpq-dev && \
+    apt-get install -y libpq-dev libclang-3.9-dev clang-3.9 \
```
Comment (guilload) [inline build/cross-images/aarch64-unknown-linux-gnu.dockerfile:7]: nit: can we have one dependency per line and sort the dependencies

File: build/cross-images/x86_64-unknown-linux-gnu.dockerfile [hunk 3]
```diff
@@ -2,5 +2,6 @@ FROM quickwit/cross-base:x86_64-unknown-linux-gnu
 # See https://github.com/quickwit-inc/cross
 
 RUN apt-get update && \
-    apt-get install -y zlib1g-dev libssl-dev libpq-dev && \
+    apt-get install -y zlib1g-dev libssl-dev \
```
Comment (guilload) [inline build/cross-images/x86_64-unknown-linux-gnu.dockerfile:5]: ditto

**Review by guilload — APPROVED** [review]
So if I understood correctly, what was missing to build RocksDB is clang?

File: build/cross-images/x86_64-unknown-linux-musl.dockerfile (hunk 1 above)
Comment (guilload) [reply build/cross-images/x86_64-unknown-linux-musl.dockerfile:15]: No worries :+1:

Comment (evanxg852000) [reply build/cross-images/x86_64-unknown-linux-musl.dockerfile]: On the other hand for the dependencies yes, RocksDB need LLVM and Clang to work properly
